src/main/java/com/socialmedia/socialmediaapi/service/RelationsService.java

```diff
@@ -0,0 +1,127 @@
+                    RelationsStatus.REQUESTFORFRIEND);
+            }
+            return relation;
+        }else if(byInterest.isPresent()){
+            relation = byInterest.get();
+            if(relation.getRelationsStatus().equals(RelationsStatus.REQUESTFORFRIEND)){
+                relation.setRelationsStatus(RelationsStatus.FRIEND);
+                relationsRepo.save(relation);
+                log.debug("Relations between subscriber: '{}' and interest: '{}' was changed to '{}'.", subscriberUuid, interestUuid,
+                    RelationsStatus.FRIEND);
+                return relation;
+            }
+        }
+        relation = relationsRepo.save(Relations
+            .builder()
+            .subscriberUuid(subscriberUuid)
+            .interestUuid(interestUuid)
+            .relationsStatus(RelationsStatus.REQUESTFORFRIEND)
+            .build());
+        log.debug("Request for friendship between subscriber: '{}' and interest: '{}' saved with uuid: '{}'.", subscriberUuid, interestUuid,
+            relation.getUuid());
+        return relation;
+    }
+
+    @Transactional
+    public Optional<Relations> confirmFriendship(UUID interestUuid, UUID subscriberUuid, boolean isConfirmed) {
+        var optionalRelation = relationsRepo.findBySubscriberAndInterestUuids(interestUuid, subscriberUuid);
+        if(optionalRelation.isPresent()){
+            var relation = optionalRelation.get();
+            if(isConfirmed){
+                relation.setRelationsStatus(RelationsStatus.FRIEND);
+            }else{
+                relation.setRelationsStatus(RelationsStatus.FOLLOWER);
+            }
+            var updatedRelation = relationsRepo.save(relation);
+            log.debug("Friendship between subscriber: '{}' and interest: '{}' with uuid: '{}' was updated successfully.", subscriberUuid,
+                interestUuid, updatedRelation.getUuid());
+            return Optional.of(updatedRelation);
+        }
+        return optionalRelation;
+    }
+
+    @Transactional
+    public Optional<Relations> removeFromFriend(UUID userUuid1, UUID userUuid2) {
+        var optionalRelation = relationsRepo.findBySubscriberAndInterestUuids(userUuid1, userUuid2);
+        if (optionalRelation.isEmpty()){
+            optionalRelation = relationsRepo.findBySubscriberAndInterestUuids(userUuid2, userUuid1);
+        }
+        if (optionalRelation.isPresent()){
+            var relation = optionalRelation.get();
+            relation.setRelationsStatus(RelationsStatus.DECLINED);
+            log.debug("Friendship between '{}' and '{}' with uuid: '{}' was removed.", userUuid1, userUuid2, relation.getUuid());
+            return Optional.of(relation);
+        }
+        return optionalRelation;
+    }
+
+    public Optional<List<UUID>> getFriends(UUID userUuid) {
+        Optional<List<UUID>> interestUuidsBySubscriberUuid = relationsRepo.findInterestUuidsBySubscriberUuid(userUuid, RelationsStatus.FRIEND);
```

Comment:
Optional of List - it's a wrong return type.
you can always return an empty list